### PR TITLE
Pptp 2538

### PR DESCRIPTION
**Pull request by mauro-fanton**
### Description of Work carried through

Brief description of what/why/how.

### CODE REVIEW

File: app/models/amends/AmendNewAnswerType.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+
+package models.amends
+
+import play.api.libs.json.{Json, OFormat}
+
+
+sealed trait  AmendNewAnswerType{
+  val value: Option[String]
+  val hiddenMessage: String
```
Comment (liampan) [inline app/models/amends/AmendNewAnswerType.scala:24]: I dont think either of these should be in this trait, that way you wont have to override it to be ""

File: app/models/amends/AmendNewAnswerType.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+case class AnswerWithValue(value: Option[String]) extends AmendNewAnswerType {
```
Comment (liampan) [inline app/models/amends/AmendNewAnswerType.scala:26]: is value optional? AnswerWithValue implies its not? 😕

Comment (mauro-fanton) [reply app/models/amends/AmendNewAnswerType.scala:26]: Yes, value could be null if we are not in amending mode

File: app/models/amends/AmendNewAnswerType.scala (hunk 1 above)
Comment (mauro-fanton) [reply app/models/amends/AmendNewAnswerType.scala:24]: happy to review this together as I battle a little bit with scala object and classes

**Review by muninderadavelly — APPROVED** [review]

**Review by liampan — APPROVED** [review]
Please add a meaningful title to the PR

**Pull request merged**